src/python/datapreprocessor/datapreprocessor/personclient.py

```diff
@@ -0,0 +1,33 @@
+from ipersonclient import IPersonClient
+
+class PersonClient(IPersonClient):
+    def get_all(self):
+        response = self._get(self._build_url('people'))
+        items = response.json()['data']
+        return items
+
+
+    def get_by_name(self, person_name):
+        people = self.get_all()
+        person = filter((lambda p: p['attributes']['name'] == person_name), people)
+        if person:
+            return person[0]
+        else:
+            return None
+
+
+    def add(self, person_name):
+        if self.get_by_name(person_name):
+            return None
+        else:
+            data = {'data': {'type': 'people', 'attributes': {'name': person_name}}}
+            response = self._post(self._build_url('people'), data)
+            return response
```

Comment:
Similar to the comment I made on the get_all function I'm not sure we should be returning the response here. When a new person's been added surely the more useful thing to return is just the id of the newly created person.